### ISSUE TITLE
Obliterates generation times, affecting Chunky heavily

### ISSUE THREAD

**Issue by Barerock** [WunderFunct/WFCaveOverhaul#13]
Ironically pregenning becomes more necessary with this mod active because it destroys chunk loading times, but with Chunky and CO active, Chunky will take upwards of THREE DAYS to generate only 10k radius, which is insane considering I can generate 10k without this mod in merely 12 hours with an otherwise identical mod set.

Optimization is needed desperately to make this wonderful generation feasible to play with. I noticed that this mod, on vanilla, will increase chunk loading to easily > 2x normal times. It's abysmal, unfortunately. I look forward to better performance.

For instance, my CPU usage with Chunky running is capped at 89%, averaging around 45-55% for 36 hours. Without CO, the average is 65-70% and capped at 98%.

**Comment by Barerock** [comment]
And yes, that Chunky metric is accurate, I lost a survival world I had just pregenned for literally four days, off for about 16 hours, so 3 days total hours. I lost it due to incompatibility with lithosphere, though previous tests I did confirmed that this mod hinders generation far more than normal in a clean environment.

**Comment by WunderFunct** [comment]
Noted. I'm sorry about you losing your world! That's a bitter pill to take. 

When did you download CO? I added an aquifer patch to fix some water-related issues in 1.3.2, but I think you mentioned in the lithosphere issue that you're on 1.3? If you're on 1.3.2 specifically, you can turn the aquifer patch off in the config since it's an experimental feature and I anticipate a fair amount of lag to be generated from it. 

If you're not running on 1.3.2, then there's definitely an environment issue going on here on my side. I've heard reports from multiple people of lag spikes and other performance issues. What's weird is my dev environment doesn't seem to have those issues. It'll take a lot of work to figure out what's going on, but I'm worried that it could be stemming from my noise library, which might or might not be optimized.

Can you confirm that you are running 1.3/1.3.1 and not 1.3.2?

**Comment by Barerock** [comment]
Just last week, and it seems 1.3.2.1 was the latest version for forge on curseforge for 1.20.1, so that'd be the one, since I don't usually test versions for compatibility unless I absolutely must.

I think a large portion of the lag in generation is due to the insane height everything generated at in previous issue. It's not nearly as performance heavy without Lithosphere active, but only by about 30 or 40%. It definitely hits hard, though I'd be willing to tank that time sink into pregen for the cavey-ness of this generation, since I can just generate overnight for a couple weeks, which is a bummer, but probably not NEARLY as bad as the 120 height bug was.

**Comment by Barerock** [comment]
I don't actually get any lag spikes either, since you mentioned it. I do have a rather beefy computer for MC though, so I could be an outlier.

**Comment by WunderFunct** [comment]
1.3.2.1 contains a new piece of code designed to fix water carving interactions. It's called the aquifer patch and it's enabled by default. It could be the culprit. I know it was giving me some performance issues, but they weren't that bad on my computer. I'll turn it off by default in the next update, but you can turn it off in the configs for the time being.

Update: I've tried running Lithosphere with CO and did see some extensive performance issues. I'll have to debug why Lithosphere and CO are acting this way as a background project. Lithosphere may be doing more noise-related worldgen attempts when compared to Vanilla, which means that CO is running multiple times. Note that this would be on my side to address if this theory is what's actually happening. I might be able to add some sort of caching to fix this issue.

**Comment by Barerock** [comment]
The latest beta build on Curseforge for 1.20.1, 1.3.2 eliminates the ocean issue and also _seems_ to generate much much faster. I will fly around for a bit to test another bug report elsewhere to see how fast compared to normal. Ocean is generating rapidly for how much _stuff_ I have installed.

**Comment by Barerock** [comment]
Yeah, mountains and whatnot? Dude this is performaning _great_ with CO

**Comment by WunderFunct** [comment]
I'm really glad to hear! I fundamentally re-wrote how certain parts of CO are done but I didn't think it'd impact performance in a meaningful way. Let me know if you're still having any issues in the future. :)

**Comment by VeggeieDude** [comment]
Ooh, does this work for 1.3.2 for fabric 1.20.1 too?
